sklearn/metrics/tests/test_common.py

```diff
@@ -2390,6 +2434,69 @@ def test_array_api_compliance(metric, array_namespace, device, dtype_name, check
     check_func(metric, array_namespace, device, dtype_name)
 
 
+@pytest.mark.parametrize(
+    "array_input, reference",
+    [
+        pytest.param(*args[:2], id=args[2])
+        for args in yield_mixed_namespace_input_combinations()
+    ],
+)
+@pytest.mark.parametrize("metric_name", sorted(METRICS_SUPPORTING_MIXED_NAMESPACE))
+def test_mixed_namespace_input_compliance(metric_name, array_input, reference):
+    """Check 'everything' follows `y_pred` for mixed namespace inputs.
+
+    If output is array, checks it is of the same namespace and device as `y_pred`
+    (`reference`).
+    Note `sample_weight` not set.
+    """
+    xp_ref = _array_api_for_tests(reference.xp, reference.device)
+    xp_input = _array_api_for_tests(array_input.xp, array_input.device)
+    all_metrics_include_pairwise = {
+        **ALL_METRICS,
+        **PAIRWISE_METRICS,
+    }
+    metric = all_metrics_include_pairwise[metric_name]
+
+    if metric_name in CLASSIFICATION_METRICS:
+        # These should all accept binary label input as there are no
+        # `CLASSIFICATION_METRICS` that are in `METRIC_UNDEFINED_BINARY` and are NOT
+        # `partial`s (which we do not test for in array API)
+        y1 = xp_input.asarray([0, 0, 1, 1], device=array_input.device)
+        y2 = xp_ref.asarray([0, 1, 0, 1], device=reference.device)
+    elif metric_name in {**CONTINUOUS_CLASSIFICATION_METRICS, **CURVE_METRICS}:
+        if metric_name not in METRIC_UNDEFINED_BINARY:
+            # Continuous binary input
+            y1 = xp_input.asarray([0.5, 0.2, 0.7, 0.6], device=array_input.device)
+            y2 = xp_ref.asarray([1, 0, 1, 0], device=reference.device)
+        else:
+            # Continuous but shape (n_samples, n_labels)
+            y1 = xp_input.asarray([[0.5, 0.2, 0.7, 0.6]], device=array_input.device)
+            y2 = xp_ref.asarray([[1, 0, 1, 0]], device=reference.device)
+    elif metric_name in REGRESSION_METRICS:
+        y1 = xp_input.asarray([2.0, 0.1, 1.0, 4.0], device=array_input.device)
+        y2 = xp_ref.asarray([0.5, 0.5, 2, 2], device=reference.device)
+    elif metric_name in PAIRWISE_METRICS:
```

Comment:
Not familiar with metric types, I *think* I got this right but open to improvements!